### ISSUE TITLE
Add ability to assign due dates to issues.

### ISSUE THREAD

**Issue by markdorison** [isaacs/github#432]

**Comment by cirosantilli** [comment]
Possible workaround: milestones. True, not perfect as too much overhead for creating one milestone per issue.

Note however that GitHub is highly resistant to new issue metadata ;-)

**Comment by markdorison** [comment]
Milestones with due dates work great in many cases but there are others
where they don't work at all, especially when the due dates all vary. One
milestone per issue doesn't really scale.

**Comment by kevinisaac** [comment]
Any luck with this feature? We would love to have this feature for our team to assign some kind of priority/deadlines for issues!

**Comment by xolotl** [comment]
I notice GitLab has due dates for issues: https://docs.gitlab.com/ee/user/project/issues/due_dates.html

Milestones is not a great solution for granular date-based tracking. It would be great if issues had due dates.

**Comment by shaform** [comment]
Since we have projects/cards, it would be great if we could assign due date to cards as well. Perhaps there should also be some notification mechanism when the task is due.

**Comment by charlie-quinn** [comment]
I have been waiting for this feature for years – now is the time GitHub! Now is the time. 🏆

**Comment by webhealer** [comment]
That would be a great option if we can assign due date for issues

Milestone not helps at all.

**Comment by crypticsymbols** [comment]
This is a huge hole in GitHub issues usefulness for Real Software™. There are many cases (SSL cert expiration, third-party service subscriptions, etc) that need regular attention on a calendar basis and there is currently no way to reflect that. We are currently considering switching our issue tracking for this sole reason.

**Comment by OldhamMade** [comment]
Agreed. Due dates for issues with email notifications when the due date is getting close help massively in projects – think SSL certificate renewals, etc.

**Comment by smith11235** [comment]
Please.

The only other thing that we ever note as a team now is that whoever made projects and whoever made milestones don't seem to get along.

**Comment by chrisblakley** [comment]
I also just noticed this feature in GitLab and would *love* to have it in Github.

**Comment by charlex** [comment]
My team has repeatedly reported that the lack of deadlines is the their number one pain point with GH issues.

**Comment by mberglund99** [comment]
Our team's largest problem with Github - no due dates on issues. Would be amazing.

**Comment by TonyRutherford1** [comment]
Is this coming? This is such a basic but key feature :-)

**Comment by YesThat** [comment]
Four years and zero acknowledgement by Github staff..  I somehow doubt that it's coming.  I do so hope, however..

**Comment by alexleventer** [comment]
I built this GitHub action to manage due dates on issues: https://github.com/alexleventer/github-issue-due-dates-action

It will tag the issues with an overdue/1 week to due labels.

**Comment by threewordphrase** [comment]
@alexleventer FTW!

But if you're listening, GitHub, don't burden this guy with the torch forever!

**Comment by brucebrit** [comment]
Please include this, GitHub. it will be the best gift for year 2020

**Comment by joshimhoff** [comment]
+1

**Comment by MitchelKerckhaert** [comment]
Could someone from the github team please respond to this issue? this is indeed a very common feature that has repeatedly been requested for almost five years now. If there is a reason for not wanting to / being able to implement this, please share in this issue.

**Comment by duythanhvn** [comment]
I hope to have an answer on this issue as soon as possible.

**Comment by OldhamMade** [comment]
Any news on this feature @duythanhvn? I know it might seem somewhat bad form to chase up with everything that is going on at the moment, but this feature is something that would really benefit teams using GH Issues for project management, _especially_ now that teams are having to work remotely.

**Comment by duythanhvn** [comment]
Sadly, I don't know @OldhamMade. I don't even know if there are employees or managers at Github and Microsoft who read these feedback or are planning to develop this feature. I'm still waiting like the other programmers here. If there's an official statement from Github then I think that's great.

**Comment by nickmayn** [comment]
does anyone know if this issue has been added? Would really like it to be included

**Comment by maartenfv** [comment]
How can we get more attention for this issue?

**Comment by chrisblakley** [comment]
> How can we get more attention for this issue?

[Github Satellite](https://githubsatellite.com/schedule/) is coming up next week. Maybe we could ask at appropriate Q&A sessions?

**Comment by havesomeleeway** [comment]
@chrisblakley did you manage to pose the question during Github Satellite?

**Comment by johann-petrak** [comment]
This would be a very fundamental and important feature to have, gitlab does have it and the issue has been open and unadressed for 5 years now. It is disappointing to see no progress on this. 

If this ever gets added it should be in parallel with due dates on project cards, especially since cards can be converted to issues (which could then have the same due date).

**Comment by danieleldjarn** [comment]
This feature is something my team is sorely missing from GitHub issues!

**Comment by MarcStdt** [comment]
We've just migrated all of our projects from Gitlab to GitHub and I was surprised about the fact, that GitHub does not have this feature.
It seems quite fundamental for project management.

**Comment by JFQueralt** [comment]
+1
We are also bummed by this.

**Comment by GAFelton** [comment]
Please implement due dates on issues! I think its an essential feature for any project.

**Comment by chrisblakley** [comment]
For what it's worth, I woke up to this notification this morning (from the Github Universe 2020 virtual conference):
![Screen Shot 2020-12-09 at 8 18 09 AM](https://user-images.githubusercontent.com/6349678/101634923-50c93100-39f7-11eb-9c47-4fcdf86db123.png)

**Comment by ChrisSLR** [comment]
GitLab and GitHub look pretty much the same until it comes to due dates. I frequently use them on GitLab to remind me of important tasks that need to be worked on later. As GitHub milestones already support due dates I don't understand why enabling it for issues is such a big deal ...

If improved usability doesn't convince you, then make it available only for paid accounts. I'm sure that many people would love to upgrade just to have this features.

**Comment by ARambazamba** [comment]
Please take a look at this request again and implement due dates ... after all not a big thing to do

**Comment by IAmSarthakVerma** [comment]
This is the only feature that feels like is MISSING from the issues tab. Deadlines are a must!

**Comment by johann-petrak** [comment]
No response by github since 2015, does not look like they would care much at all about feature requests by their users.

**Comment by renestefancic** [comment]
Is there an ETA on this?

**Comment by ARambazamba** [comment]
> This is the only feature that feels like is MISSING from the issues tab. Deadlines are a must!

... except „create branch for issue“ … 🙈

**Comment by cspavlou** [comment]
+1, this is a feature that we would love to see in GitHub

**Comment by TonyRutherford1** [comment]
By the looks of it.....Never going to happen 😭

**Comment by cnyyk** [comment]
Now is 2021-03-23, is this coming? This is such a basic but key feature :-)

**Comment by andreas-bergstrom** [comment]
Pretty obvious that MS wants Github to stay light since they also sell Azure Devops. We are probably migrating to Gitlab instead.

**Comment by ShixiangWang** [comment]
Use https://github.com/marketplace/actions/due-dates

**Comment by FedeCuci** [comment]
+ 1

**Comment by Gustavo-Henrique-EM** [comment]
.+1

**Comment by khuderm** [comment]
This issue was created 6 years ago and no response from github. Perhaps it is time to move on to gitlab😡

**Comment by Levi-Lesches** [comment]
@khuderm, GitHub doesn't monitor this repository. Use [the official GitHub feedback discussions](https://github.com/github/feedback/discussions/categories/general-feedback) if you want a response.

**Comment by Ahacad** [comment]
Some of the discussions here: https://github.com/github/feedback/discussions/4282
